hive/src/test/java/com/netease/arctic/hive/catalog/HiveCatalogLoaderTest.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netease.arctic.hive.catalog;
+
+import com.netease.arctic.TestAms;
+import com.netease.arctic.ams.api.CatalogMeta;
+import com.netease.arctic.ams.api.properties.CatalogMetaProperties;
+import com.netease.arctic.ams.api.properties.TableFormat;
+import com.netease.arctic.catalog.ArcticCatalog;
+import com.netease.arctic.catalog.CatalogLoader;
+import com.netease.arctic.catalog.CatalogTestHelpers;
+import com.netease.arctic.hive.TestHMS;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import java.util.Map;
+
+public class HiveCatalogLoaderTest {
+  @ClassRule
+  public static TestAms TEST_AMS = new TestAms();
+
+  @ClassRule
+  public static TestHMS TEST_HMS = new TestHMS();
+
+  private static final String TEST_CATALOG_NAME = "test";
```

Comment:
if TEST_HMS is a @ClassRule, the HMS will be created and destroyed at each test class.

Reply:
I add a new system property `singleton-test-resource` to support only creating one test resource for all test cases. You can find related codes in class `SingletonResourceUtil` 、`TestAms` and `TestHMS`.
I plan to set `singleton-test-resource` to `true` in our Github CI execution to reduce resource creating cost.